### PR TITLE
[Accessibility] Adjust color sales badge color contrast

### LINKED ISSUE #51895: Accessibility: Color contrast insufficient
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The sale text on the product images in the "You may be interested in..." section fails color contrast against the background color. See color contrast analysis: https://webaim.org/resources/contrastchecker/?fcolor=B3AF54&bcolor=FFFFFF

Image

### Component or Page: 

Shortcode: [woocommerce_cart] & Cart Classic Block

### Name affected element:

"Sale" text

### Relevant code snippets:

`<span class=""onsale"">Sale!</span>

.woocommerce span.onsale {
    background-color: var(--wp--preset--color--foreground, #b3af54);
    color: var(--wp--preset--color--background, #fff);
}`

### Recommended fix:

Change the colors to have a minimum color contrast of at least 4.5:1.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #51895 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure your Cart page uses the classic variation.
1. Select a product (Product A) and add a sales price to it.
image
2. Select another product (Product B) and open its admin page.
3. On the admin page, go to "Linked Products" > "Cross-sell" and add Product A as a cross-sell of Product B.
image
4. On the frontend, add Product B to the cart.
5. Go to the Cart page.
6. Scroll down. The Product A should be in the "You may be interested in..." section.
7. Verify the sales badge of Product A has sufficient contrast between background and foreground colors.
8. Perform the test on WP default themes and Storefront theme.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/legacy/css/_variables.scss [hunk 1]
```diff
@@ -14,8 +14,8 @@ $primarytext:       desaturate(lighten($primary, 50%), 18%) !default;    // Text
 $secondary:         desaturate(lighten($primary, 40%), 21%) !default;    // Secondary buttons
 $secondarytext:     desaturate(darken($secondary, 60%), 21%) !default;   // Text on secondary color bg
 
-$highlight:         adjust-hue($primary, 150deg) !default;               // Prices, In stock labels, sales flash
-$highlightext:      desaturate(lighten($highlight, 50%), 18%) !default;  // Text on highlight color bg
+$highlight:         adjust-hue(darken($primary, 18%), 150deg) !default;  // Prices, In stock labels, sales flash
+$highlightext:      desaturate(lighten($highlight, 80%), 18%) !default;  // Text on highlight color bg
```
Comment (Manussakis) [inline plugins/woocommerce/client/legacy/css/_variables.scss:18]: Adjustment for Twenty Twenty-Four theme.
**Before**
image

**After**
Screenshot 2024-10-16 at 18 19 03

File: plugins/woocommerce/client/legacy/css/twenty-twenty-one.scss [hunk 2]
```diff
@@ -8,7 +8,7 @@ $headings: var(--heading--font-family);
 $body: var(--global--font-secondary);
 
 $body-color: currentColor;
-$highlights-color: #88a171;
+$highlights-color: var(--wc-highlight, #777335);
```
Comment (Manussakis) [inline plugins/woocommerce/client/legacy/css/twenty-twenty-one.scss:11]: Adjustment for Twenty Twenty-One theme.

**Before**
Screenshot 2024-10-16 at 17 55 49

**After**
Screenshot 2024-10-16 at 17 56 50

**Comment by Manussakis** [comment]
@dkotter This task is ready for code review.
Thanks!

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by qasumitbagthariya** [comment]
### QA Update  ⚠️
---

@Manussakis 

I’ve verified the sales badge on both the Storefront and TT4 themes, and it appears differently between the two. In Storefront, the badge looks distinct from the one in TT4. Could you please check if this is expected behavior?



**Theme: TT4** 

image

image


**Storefront:**

image

image

image

**Comment by Manussakis** [comment]
@qasumitbagthariya Thanks for testing it.
Yes, this is the expected behavior. This task only fixes the badge color contrasts, not their appearance.

**Comment by qasumitbagthariya** [comment]
@Manussakis Thanks for the detailed update, I'm approving this PR.

please remove [WIP] from the title and take this PR out of draft.

**Comment by jeffpaul** [comment]
@albarin this PR is ready for your review

**Review by alexflorisca — APPROVED** [review]
Everything looks good from my side too, tested with TT - TT5 and Storefront. 👍

**Pull request merged**